src/TorchVision/dsets/CelebA.cs

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using static TorchSharp.torch;
+using static TorchSharp.torch.nn;
+using static TorchSharp.torch.utils.data;
+using static TorchSharp.torchvision.datasets.utils;
+
+// A number of implementation details in this file have been translated from the Python version of torchvision,
+// largely located in the files found in this folder:
+//
+// https://github.com/pytorch/vision/blob/b78d98bb152ffb9c0c0f5365f59f475c70b1784e/torchvision/datasets/celeba.py
+//
+// The origin has the following copyright notice and license:
+//
+// https://github.com/pytorch/vision/blob/main/LICENSE
+//
+
+#nullable enable
+namespace TorchSharp
+{
+    namespace Modules
+    {
+        internal sealed class CelebA : Dataset
+        {
+            private static readonly (string, string, string)[] file_list = new (string, string, string)[] {
+                // File ID                                      MD5 Hash                            Filename
+                ("0B7EVK8r0v71pZjFTYXZWM3FlRnM", "00d2c5bc6d35e252742224ab0c1e8fcb", "img_align_celeba.zip"),
```

Comment:
Is there any chance that these hashes will change if the dataset is updated?

Reply:
That will be changed when the content is updated, but I didn't see any changes for recent years.